packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -76,6 +77,75 @@ function createDetachableIframe(): HTMLIFrameElement {
     return iframe;
 }
 
+interface BrowserEnvironmentOptions {
+    distortionCallback?: DistortionCallback;
+    endowments?: PropertyDescriptorMap;
+    keepAlive?: boolean;
+    instrumentation?: Instrumentation | undefined;
+}
+
+interface RealmController {
+    cleanup: () => {};
+    evaluator: typeof eval;
+    getGlobalOwnKeys: () => (string | symbol)[];
+    needsExplicitCleanUp: boolean;
+}
+// @ts-ignore: TypeScript doesn't know what a ShadowRealm is yet.
+const { ShadowRealm: ShadowRealmCtor } = globalThis;
+function createRealmController(options: BrowserEnvironmentOptions): RealmController {
+    let cleanup = () => {};
+    let evaluator: typeof eval;
+    let getGlobalOwnKeys: () => (string | symbol)[];
+    let needsExplicitCleanUp = false;
+
+    if (typeof ShadowRealmCtor === 'function') {
+        // @ts-ignore: TypeScript doesn't know what a ShadowRealm is yet.
+        const shadowRealm = new ShadowRealm();
+        evaluator = shadowRealm.evaluate.bind(shadowRealm);
+        const getGlobalThisOwnKeysFromShadowRealm = evaluator(`(callableKeysCallback) => {
+        let ownKeys = Reflect.ownKeys(globalThis);
```

Comment:
We need `ReflectOwnKeys` and `ReflectApply` references

Reply:
Inside the newly created shadow realm?

Reply:
Ah I see.  I missed that context.

Reply:
Added a comment about it!